beep/structure.py

```diff
@@ -527,6 +528,12 @@ def from_arbin_file(cls, path, validate=False):
         metadata_path = path.replace(".csv", "_Metadata.csv")
         data = pd.read_csv(path)
         data.rename(str.lower, axis='columns', inplace=True)
+
+        for column, dtype in ARBIN_CONFIG['data_types'].items():
+            if column in data:
+                if not data[column].isnull().values.any():
+                    data[column] = pd.Series(data[column], dtype=dtype)
```

Comment:
This is skipping the type of an entire column if a single value is bad. You probably don't need the "if" to check for null values, and can just convert directly? Also, you can also change datatype using data[column].astype(dtype). Not sure about memory requirements of that vs your version.

Reply:
The problem is that NaN values are floats so if a column is missing some values and we set those to NaN, and then try to convert the data type to `int` there will be an error. The `if` statement is to address that situation and just leave the data column as its current type.
Thanks for spotting the `pd.Series`. I will switch over to the other method for type conversion, I think it's more consistent with the rest of the code.

Reply:
I see -- I've only tried float64 to float32. Good to know about the issue with int.